### PR TITLE
feature: Optimized pubsub api demo

### PR DESCRIPTION
**Pull request by gdutChenxj**
**What this PR does**:
mix pubsub demo

**Which issue(s) this PR fixes**:

Fixes [#584](https://github.com/mosn/layotto/issues/584)

### CODE REVIEW

**Comment by gdutChenxj** [comment]
- i have  signed CLA but the status is not sync automatically
-  to merge all pubsub demo, some folder will be deleted and it make  an error when `Quickstart Validation`  exec cmd `cd /home/runner/work/layotto/layotto/demo/pubsub/redis/server/`，the script seem to be build on the workflow server，i think it need somebody to update the script or ignore the validation error

**Comment by seeflood** [comment]
>i have signed CLA but the status is not sync automatically

Fixed. I closed and reopened this PR to rerun the workflow .

>to merge all pubsub demo, some folder will be deleted and it make an error when Quickstart Validation exec cmd cd /home/runner/work/layotto/layotto/demo/pubsub/redis/server/，the script seem to be build on the workflow server，i think it need somebody to update the script or ignore the validation error

Yes you are right :)
The quickstart doc http://localhost:3000/#/en/start/pubsub/start  and http://localhost:3000/#/zh/start/pubsub/start used the redis demo:
image
image
And Layotto runs these quickstart document in the workflow, see http://localhost:3000/#/zh/development/test-quickstart

So, could u help modify the documentation in this PR?
Just modify the doc to use the new demo code.
If you don't know how to fix it after reading the doc, I can help submit a PR to your branch to fix it :)

**Comment by gdutChenxj** [comment]
i will update the doc for the new demo

**Comment by gdutChenxj** [comment]
@seeflood  I have updated the quick start doc and it can work well in my local test. But there is a confliction need somebody with write access to handle.

**Comment by seeflood** [comment]
> @seeflood I have updated the quick start doc and it can work well in my local test. But there is a confliction need somebody with write access to handle.

Thanks ! I just saw this message.
You have the permission to resolve the conflicts.
To resolve them, you can:
1.  Merge main branch into your branch, on your computer. The IDE will guide you to fix conflicts
2. After all resolved and merged, you just push your new code to your github repo.

File: docs/en/start/pubsub/start.md [hunk 1]
```diff
@@ -48,14 +48,14 @@ Parameter Description:
...
- cd ${project_path}/demo/pubsub/redis/server/
+ cd ${project_path}/demo/pubsub/server/
  go build -o subscriber
  ```
 
 Start subscriber:
 
 ```shell @background
- ./subscriber
+ ./subscriber -s redis
```
Comment (seeflood) [inline docs/en/start/pubsub/start.md:58]:
```suggestion
 ./subscriber -s pub_subs_demo
```

File: docs/en/start/pubsub/start.md [hunk 2]
```diff
@@ -98,9 +98,9 @@ After completion, the layotto file will be generated in the directory, run it:
 ### Step 4. Run the Publisher program and call Layotto to publish events
 
 ```shell
- cd ${project_path}/demo/pubsub/redis/client/
+ cd ${project_path}/demo/pubsub/client/
  go build -o publisher
- ./publisher
+ ./publisher -s redis
```
Comment (seeflood) [inline docs/en/start/pubsub/start.md:103]:
```suggestion
 ./publisher -s pub_subs_demo
```

File: docs/zh/start/pubsub/start.md [hunk 3]
```diff
@@ -45,12 +45,12 @@ docker run -itd --name redis-test -p 6380:6379 redis
...
 ### 第二步：启动Subscriber程序,订阅事件
 ```shell
- cd ${project_path}/demo/pubsub/redis/server/
+ cd ${project_path}/demo/pubsub/server/
  go build -o subscriber
 ```
 
 ```shell @background
- ./subscriber
+ ./subscriber -s redis
```
Comment (seeflood) [inline docs/zh/start/pubsub/start.md:53]:
```suggestion
 ./subscriber -s pub_subs_demo
```

File: docs/zh/start/pubsub/start.md [hunk 4]
```diff
@@ -95,9 +95,9 @@ go build -o layotto
 ### 第四步：运行Publisher程序，调用Layotto发布事件
 
 ```shell
- cd ${project_path}/demo/pubsub/redis/client/
+ cd ${project_path}/demo/pubsub/client/
  go build -o publisher
- ./publisher
+ ./publisher -s redis
```
Comment (seeflood) [inline docs/zh/start/pubsub/start.md:100]:
```suggestion
 ./publisher -s pub_subs_demo
```

**Review by seeflood — CHANGES_REQUESTED** [review]

File: demo/pubsub/README.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+You can run server/client demo with different component names.   
+It is worth noting that both server and client demo should set the same store name by param `-s`.  
+For example:   
+```go
```
Comment (seeflood) [inline demo/pubsub/README.md:4]:
~~~suggestion
```shell
~~~

File: demo/pubsub/README.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+cd ${project_path}/demo/pubsub/server/
+// 1. start subscriber
+go build -o subscriber
+/.subscriber
```
Comment (seeflood) [inline demo/pubsub/README.md:8]:
```suggestion
/.subscriber -s pub_subs_demo
```

File: demo/pubsub/README.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+// 2. start layotto
+cd ${project_path}/cmd/layotto
+go build -o layotto
+./layotto start -c ../../configs/config_in_memory.json
+
+// 3. start publisher
+ cd ${project_path}/demo/pubsub/client/
+go build -o publisher
+ ./publisher
```
Comment (seeflood) [inline demo/pubsub/README.md:18]:
```suggestion
 ./publisher -s pub_subs_demo
```

**Review by seeflood — CHANGES_REQUESTED** [review]

**Comment by gdutChenxj** [comment]
@seeflood all conversations are resolved

**Review by seeflood — APPROVED** [review]
lgtm. Thanks for your contribution and welcome to the community !

**Review by zhenjunMa — APPROVED** [review]

**Pull request merged**